ipc/api/src/cross.rs

```diff
@@ -18,12 +21,15 @@ pub struct IpcEnvelope {
     /// It makes sense to extract from the encoded message
     /// all shared fields required by all message, so they
     /// can be inspected without having to decode the message.
+    #[serde_as(as = "HumanReadable")]
     pub to: IPCAddress,
     /// Value included in the envelope
     pub value: TokenAmount,
     /// address sending the message
+    #[serde_as(as = "HumanReadable")]
     pub from: IPCAddress,
     /// abi.encoded message
+    #[serde_as(as = "serde_with::hex::Hex")]
```

Comment:
Does this affects the actual encoding even inside IPC applications? Will it turn every byte sequence into a `String`, doubling it in size, even if you encode the message as CBOR? Don't you want to do the `HumanReadable` trick to only do `Hex` when the target is JSON or other text format?